### ISSUE TITLE
Expand flux-shell options for cpu-affinity

### ISSUE THREAD

**Issue by vsoch** [flux-framework/flux-core#4874]
There is some brief documentation here:

https://flux-framework.readthedocs.io/projects/flux-core/en/latest/man1/flux-mini.html?highlight=cpu-affinity#shell-options

But it would be good to expand this, and possibly have it somewhere written out more in full to walk the reader through the differences. E.g., start with what the job shell is/ does - it's responsible for managing and running jobs. There are options you can set that achieve <X>. Then talk about what CPU affinity is  - "an OS thing" that... `<insert explanation>`. And explain that process (which I don't understand) and then explain the different affinity arguments in that context:

- The default mode for Flux is probably most like --bind-to none 
- By default each task is bound to all the cores assigned to the job.
- Use -o cpu-affinity=per-task to bind tasks to cores by default, "core" binding (when would we want to do this and why?)
 
Thanks to @grondo for the above I'm mostly just the translator!

**Comment by grondo** [comment]
FWIW, the job shell is described in https://flux-framework.readthedocs.io/projects/flux-core/en/latest/man1/flux-shell.html (the flux-shell manpage). The long description of what CPU affinity is might be better in a FAQ entry, or just point to a standard reference elsewhere.

**Comment by vsoch** [comment]
Yep that's what I was thinking - should we transfer this to flux-docs? And maybe we can talk about a strategy for organizing content specific to the libraries like this - I suspect our FAQ page is going to get very large if that's the right place for all these details!
